Refactored Disabled component to use React Hooks

javidalkaruzi opened the pull request:
## Description
See #22890

Refactored the Disabled component to use React Hooks.

## How has this been tested?
Tested using existing unit tests.
Observed in Storybook that the components specified were disabled when rendered and automatically disabled again if I, the user, manipulate the DOM

## Types of changes
Code refactor

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [ ] My code follows the accessibility standards.
- [ ] My code has proper inline documentation.
- [ ] I've included developer documentation if appropriate.
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

Commit:
Refactored Disabled component to use React Hooks

Commit:
Refactored Disabled component to use React Hooks

spacedmonkey committed:
Image Editor: Check for upload_files capability (#23423)

Co-authored-by: Timothy Jacobs <timothy@ironbounddesigns.com>

youknowriad committed:
run jobs on fork PRs (#23538)

ockham committed:
REST API: Sync /themes endpoint with Core's (#23321)

In #21578, I added a few fields to Core's `/themes` endpoint, for use by the Site Editor's Template Switcher (see both #21578 and #21768). I then submitted those changes as a [PR](https://github.com/WordPress/wordpress-develop/pull/222) against Core. That PR underwent a number of modifications and was eventually merged; the new fields will be part of the `/themes` endpoint exposed by Core starting from the next WP release.

This PR updates the fields added by Gutenberg to follow the same semantics, as well as the callsites that use that endpoint.

ajitbohra committed:
Docs: link codepen example (#23534)

glendaviesnz committed:
Classic block: save content bug fix (#23408)

* trigger content attribute updates when classic block content changes

Co-authored-by: Glen Davies <glen.davies@a8c.com>

TimothyBJacobs committed:
Image Editor: Stabilize endpoint (#23536)

* Update the image editor route to be wp/v2/media/<id>/edit.

* Check that mime type is valid.

* Use request object directly.

* Include error message in snackbar notice.

* Update lib/class-wp-rest-image-editor-controller.php

* Use the "edit" context in the request for the new media item.

* Slash new attachment data.

* Make file type check error message more clear.

* Fix file header while we're at it.

Co-authored-by: Andrew Ozz <743931+azaozz@users.noreply.github.com>

spacedmonkey committed:
Add filter to REST API image edit. (#23539)

* Add action to REST API image edit.

* Change param on action.

* Convert to filter.

StevenDufresne committed:
Add svg icon support to Block Directory search result item. (#23442)

mkaz committed:
Remove errant </a> tag (#23542)

youknowriad commented on packages/components/src/disabled/index.js:
```diff
@@ -31,40 +36,11 @@ const DISABLED_ELIGIBLE_NODE_NAMES = [
-	constructor() {
-		super( ...arguments );
+function Disabled( { className, children, ...props } ) {
+	const node = useRef( null );
```
```suggestion
	const node = useRef();
```

I don't think it makes any difference but was wondering why setting it explicitly here.

javidalkaruzi replied:
I don't think it made any difference.

youknowriad commented:
Great refactoring, thanks. Can you rebase the PR to fix the checks?

talldan committed:
Try using a percentage based calculation for block drag scrolling velocity. (#23448)

* Use a percentage based calculation for scrolling when dragging blocks

* Factor in scroll container offset

jasmussen committed:
Fix close button. (#23551)

* Fix close button.

Close buttons across the block editor uses the x icon. This PR brings that consistency back to the recently improved post-publish panel.

* Update snapshots.

desaiuditd committed:
Refactor Embed Edit component: Class component to Function component (#22846)

* Refactor Embed Edit component

Convert the Embed Edit component into a function component so that React hooks can be used within the component.

* Fixed rebase conflicts with b2ec648a6b993b8c3f5d5a4a242ec419a88d8343
  and a610923b7728f1192f9745a55390a7abb0df576e

* Update EmbedEditComponent's effect dependencies

Co-authored-by: Zebulan Stanphill <zebulanstanphill@protonmail.com>

* Fix URL-setting effect dependencies

Co-authored-by: Miguel Fonseca <miguelcsf@gmail.com>
Co-authored-by: Zebulan Stanphill <zebulanstanphill@protonmail.com>

jasmussen committed:
Retire numerous colors from the _colors.scss (#23454)

* Opacities: Remove completely unused colors.

* Comments.

* Navigation colors: Use standard colors.

* Opacities: Remap the remaining and retire.

* Retire blue-dark-900

* Retire blue-medium-800.

* Retire more alternate colors.

* Update changelog

* Fix missing variable issue.

adamziel committed:
re-select the post for each serializedCallback, not just the queued ones (#23559)

lukewalczak committed:
[RNMobile] Some tweaks in block picker (#23465)

SergioEstevao committed:
Issue 331/mention support on @ keypress (#22119)

* Add '@' button for inserting mentions on mobile

* Move react-native-azted dependency to external dependencies section

* Handle promise rejection.

* Fix focus issue.

* Add space after mention

* Update selection onFocus.

* Check for site capabilities for mentions support

* Add the mention button inside a toolbar.

* Use HOC for site capabilities.

* Only include space after mention on iOS

On Android we are discarding selections when we think they might get
stripped by Aztec, so adding the space to the ends results in the
selection values getting discarded.

* Use onKeyDown instead of onEnter and onBackspace.

* Intercept @ keypress to trigger mention UI.

* Intercept @ keypress to trigger mention UI.

* Trigger the UI for mentions only when there is a space before the @

* Put mentions behind the DEV flag.

* Only trigger @ keypress on DEV builds.

* Remove DEV flag

* Only make mention keypress available when capabilities and editing menu are on.

* Enable space after mention on Android

* Remove DEV flag for toolbar mention button.

* Bring changes from gb-mobile to the monorepo structure.

* Check triggerKeyCodes on Android

* Add newline to end of file

* Update code to use keycodes.

* Update GB main reference.

* Update dependencies.

Co-authored-by: Matt Chowning <matt.chowning@automattic.com>

ellatrix committed:
Scroll on drag: remove dragElement dependency (#23561)

etoledom committed:
Fix gutenberg related warnings on xcode project (#23505)

etoledom committed:
[iOS] Adding the posibility for the client to addd a loading view (#23506)

This view is shown while the editor is loading

glendaviesnz committed:
Classic Block: fix undo keyboard shortcut (#23376)

* Prevent gutenberg undo from kicking in when in classic block
Co-authored-by: Glen Davies <glen.davies@a8c.com>

epiqueras committed:
Template Part: Improve insertion flow. (#23295)

* Template Part: Improve insertion flow.

* Template Part: Fix styles and remove unnecessary properties.

* Template Part: Update tests.

* Template Part: Improve popover design.

* Template Part: Improve name panel style.

* Fix property handling.

Commit:
Merge branch 'refactor/disabled-component' of github.com:javidalkaruzi/gutenberg into refactor/disabled-component

javidalkaruzi committed:
Update packages/components/src/disabled/index.js

Co-authored-by: Riad Benguella <benguella@gmail.com>

youknowriad commented:
Looks like you messed up the branch while rebasing? (both PRs)

javidalkaruzi commented:
> Looks like you messed up the branch while rebasing? (both PRs)

Yeah. I had rebased master into these branches.
